src/main/java/org/aksw/simba/lemming/metrics/single/MaxVertexDegreeMetric.java

```diff
@@ -29,4 +34,181 @@ public double apply(ColouredGraph graph) {
         }
     }
 
+    /**
+     * The method checks if we need to compute in degree or out-degree and then
+     * calls the metricComputationMaxDegree with correct parameters.
+     * 
+     * @param triple
+     *            - edge on which graph operation is performed.
+     * @param metric
+     *            - input metric which needs to be computed.
+     * @param graph
+     *            - input graph.
+     * @param graphOperation
+     *            - boolean value indicating graph operation. ("true" for adding an
+     *            edge and "false" for removing an edge)
+     * @param previousResult
+     *            - UpdatableMetricResult object containing the previous computed
+     *            results.
+     * @return
+     */
+    @Override
+    public UpdatableMetricResult update(TripleBaseSingleID triple, ColouredGraph graph, boolean graphOperation,
+            UpdatableMetricResult previousResult, VertexDegrees mVertexDegrees) {
+        UpdatableMetricResult newMetricResult;
+
+        if (direction == DIRECTION.in) {
+
+            if (graphOperation) { // graphOperation is true then add an edge otherwise its remove an edge
+                newMetricResult = metricComputationMaxDegree(graph, DIRECTION.in,
+                        triple.headId, triple, 1, previousResult, mVertexDegrees);
+            } else {
+                newMetricResult = metricComputationMaxDegree(graph, DIRECTION.in,
+                        triple.headId, triple, -1, previousResult, mVertexDegrees);
+            }
+
+        } else {
+            if (graphOperation) {
+                newMetricResult = metricComputationMaxDegree(graph, DIRECTION.out,
+                        triple.tailId, triple, 1, previousResult, mVertexDegrees);
+            } else {
+                newMetricResult = metricComputationMaxDegree(graph, DIRECTION.out,
+                        triple.tailId, triple, -1, previousResult, mVertexDegrees);
+            }
+
+        }
+
+        return newMetricResult;
+    }
+
+    /**
+     * The method contains logic that reduces the number of calls to apply method
+     * for the max vertex degree metric.
+     * 
+     * @param metric
+     *            - metric which should be calculated.
+     * @param graph
+     *            - input graph.
+     * @param metricName
+     *            - can be "RemoveAnEdge" or "AddAnEdge" indicating how the edge is
+     *            modified.
+     * @param direction
+     *            - this is in or out based on the operation.
+     * @param vertexID
+     *            - The vertex that is modified.
+     * @return
+     */
+    private UpdatableMetricResult metricComputationMaxDegree(ColouredGraph graph, 
+            DIRECTION direction, int vertexID, TripleBaseSingleID triple, int updateVertexDegree,
+            UpdatableMetricResult previousResult, VertexDegrees mVertexDegrees) {
```

Comment:
Please check the parameters of this method. It seems like `triple` is not used and `direction` is an attribute of this class.

Reply:
Done.